kubernetes/kops
# Update the service manifest for Docker

Updating based on latest Docker service manifest:
https://github.com/docker/docker-ce-packaging/blob/360f755c768ebb06c9974705674302098bb9c398/systemd/docker.service

This ended up being a bit bigger than expected, but mostly because a new version of `github.com/blang/semver/v4` was needed to parse the docker version, which contains leading `0`s.

Summary (with similar commit names):
* add cgroup related dependencies
* use "infinity" for LimitNOFILE and LimitNPROC for less accounting overhead
* remove workaround for old unsupported Docker versions
* remove workaround for socket activation on CentOS/RHEL distros
* rearrange manifest to match official manifest
* add containerd service dependency for newer Docker versions (with new `semver` as prerequisite)
